lua/wire/client/e2helper.lua

```diff
@@ -290,29 +302,54 @@ function E2Helper.Update()
 	local maxcount = E2Helper.MaxEntry:GetValue()
 	local tooltip = E2Helper.Tooltip:GetChecked(true)
 
-	for _, v in pairs(CurrentTable()) do
-		if E2Helper.CurrentMode == true then
-			local argnames, signature, rets, func, cost = v.argnames, unpack(v)
-			local name, args = string.match(signature, "^([^(]+)%(([^)]*)%)$")
-
-			if signature:sub(1, 3) ~= "op:" and
-					name:lower():find(search_name, 1, true) and
-					args:lower():find(search_args, 1, true) and
-					rets:lower():find(search_rets, 1, true) then
-				local line = E2Helper.ResultFrame:AddLine(name, args, rets, cost or 20)
-				if tooltip then line:SetTooltip(E2Helper.GetFunctionSyntax(name, args, rets)) end
+	-- add E2 constants
+	E2Helper.constants = {}
+	if E2Helper.CurrentMode == true then
+		for k, v in pairs(wire_expression2_constants) do
+			-- set the type according to the functions
+			local strType = type(v)
```

Comment:
`local strType = E2Lib.guess_type(v)`

Reply:
Constants can only be strings or numbers anyway, so guess type wouldn't do anything. (See https://github.com/wiremod/wire/blob/master/lua/entities/gmod_wire_expression2/base/tokenizer.lua#L195-L205)

Reply:
`E2Lib.guess_type` is going to error when given a number, that should probably be fixed.

Reply:
@Divran I'll update that so the substitution of constants for their values is done in the parser instead of the tokenizer. That way, it could create a `literal` node with type `E2Lib.guess_type(value)`, or do even better and actually specify the type of the constant in the arguments to `E2Lib.registerConstant`.

@bigdogmat fixed, 290215ab9fe82058fda166e1cd4de568079862f1

Reply:
So if I understand this correctly, use E2Lib.guess_type is fine now...?